packages/next/lib/find-pages-dir.ts

```diff
@@ -10,22 +10,37 @@ export const existsSync = (f: string): boolean => {
   }
 }
 
-export function findPagesDir(dir: string): string {
-  // prioritize ./pages over ./src/pages
-  let curDir = path.join(dir, 'pages')
+function findDir(dir: string, name: string): string | null {
+  // prioritize ./${name} over ./src/${name}
+  let curDir = path.join(dir, name)
   if (existsSync(curDir)) return curDir
 
-  curDir = path.join(dir, 'src/pages')
+  curDir = path.join(dir, 'src', name)
   if (existsSync(curDir)) return curDir
 
-  // Check one level up the tree to see if the pages directory might be there
-  if (existsSync(path.join(dir, '..', 'pages'))) {
+  return null
```

Comment:
```suggestion
  return undefined
```

Then we don't need to check it's null or undefined later?

Reply:
This one is intentional to have a strict check for the value not existing.